### PR TITLE
Table Valued Function

### PR DESCRIPTION
**Pull request by RotendaBBD**
This function returns a table of sitter application with the same status.

### CODE REVIEW

File: flyway/migrations/V17__TableValuesFunction.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CREATE FUNCTION GetSitterApplicationByStatus
+(
+    @status VARCHAR(225)
```
Comment (jennadunfordbbd) [inline flyway/migrations/V17__TableValuesFunction.sql:3]: No @  here.

Also, status is a predefined variable in MySQL, please change to something else.

File: flyway/migrations/V17__TableValuesFunction.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+CREATE FUNCTION GetSitterApplicationByStatus
```
Comment (jennadunfordbbd) [inline flyway/migrations/V17__TableValuesFunction.sql:1]: Needs `DELIMITER //` statement at top.

File: flyway/migrations/V17__TableValuesFunction.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    @status VARCHAR(225)
+)
+
+RETURNS TABLE
+AS
+RETURNS
+(
+    SELECT *
+    FROM SitterApplications
+    WHERE applicationStatus = @status
```
Comment (jennadunfordbbd) [inline flyway/migrations/V17__TableValuesFunction.sql:12]: Remove @ from status

File: flyway/migrations/V17__TableValuesFunction.sql [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+)
```
Comment (jennadunfordbbd) [inline flyway/migrations/V17__TableValuesFunction.sql:13]: Needs delimeter statement at end: 
`END;`
`//`
`DELIMITER ;`

**Review by jennadunfordbbd — CHANGES_REQUESTED** [review]
Thank you, please address comments.

**Comment by jennadunfordbbd** [comment]
Hello, MYSQL 8 does not support table value functions.

This is a working version of your sql where it is now a proc instead of a function:

```
DELIMITER //

CREATE PROCEDURE GetSitterApplicationByStatus(
    IN applicantStatus VARCHAR(225)
)
BEGIN
    SELECT * FROM SitterApplications WHERE applicationStatus = applicantStatus;
END//

DELIMITER ;
```

**Review by jennadunfordbbd — APPROVED** [review]
Thanks! All looks good :)

**Review by Lior-Becker — APPROVED** [review]

**Pull request merged**